README.md

```diff
@@ -2618,4 +2624,121 @@ React is a JavaScript library for building user interfaces. It takes the backend
+
+### HyperText Transfer Protocol (HTTP)
+
+The client to web server and back communications are in the form of Internet Protocols (IPs). Two of the more commonly known and used IPs are HyperText Transfer Protocol (HTTP) and HyperText Transfer Protocol Secure (HTTPS). HTTPS is HTTP with a layer of encryptions, which come from a Secure Socket Layer (SSL).
+
+HTTP and HTTPS always consist of a header and, depending on the method called, can also contain a body. The body can be written in JSON, HTML, JavaScript, CSS, XML etc.
+
+#### Key HTTP Methods
+
+- `GET` - Client asks web server for data.
+- `POST` - Client sends data to web server.
+- `PUT` - Client requests that web server updates data (client sends entirety of data).
+- `PATCH` - Client requests that web server updates specific data (client only sends relevant data).
+- `DELETE` - Client requests that web server deletes data.
+
+Web servers are not required to follow these methods, and can use methods (e.g. `PUT`) to action something else.
+
+When sent from the client, methods such as `POST`, `PUT` and `PATCH` always consist of a header and a body (as the client needs to send the web server the data it wants to add or update).
+
+The `GET` method does not need a body when sent from the client as additional information can be included in the Uniform Resource Locator (URL).
+
+The body is sometimes referred to as the payload. The responses from the web server will always have a body.
+
+The sender will specify how many bytes the body consists of. Generally each character is one byte, although some characters e.g. Chinese characters could be more than one byte. If the body received contains a smaller or larger number of bytes, then the receiver knows that the content could have been corrupted.
+
+### HTTP Status Codes
+
+HTTP status codes are all 3 digits long. More commonly known ones are `200` - 'OK', `202` - 'Accepted' and `404` - 'Data Not Found'.
+
+When they begin with the following numbers, status codes mean:
+
+- `1..` = hang on
+- `2..` = ok
+- `3..` = redirecting
+- `4..` = user messed up
+- `5..` = server messed up
+
+### Uniform Resource Locator (URL)
+
+The client to web server communications are enabled by web browsers. Web browsers are software applications with Graphical User Interfaces (GUIs) that take the request made by the person/computer and retrieve it from a web server using a Uniform Resource Locator (URL).
+
+URLs are signposts to web servers and usually consist of domains, resources and query parameters.
+
+The URL `https://www.amazon.com` is in the form of HTTPS and consists of three domains. `www` is a sub-domain of the `amazon` domain and `com` is the top level domain. Large companies such as Amazon will purchase all sub-domains and top level domains of their domain (e.g. `amazon`) so they can re-direct all potential customers to their main site.
```

Comment:
You don't have to buy sub-domains as these can be created when you own the domain.
```suggestion
The URL `https://www.amazon.com` is in the form of HTTPS and consists of three domains. `www` is a sub-domain of the `amazon` domain and `com` is the top level domain. It's in the interest of companies to purchase a domain name with their brand name, and will often try to purchase under as many top-level domains (such as <company>.com & <company>.io, and regional domains such as <company>.co.uk) as possible so they can re-direct all potential customers to their main site.
```